### DISCUSSION TITLE
ly2: new breakage help?

### DISCUSSION THREAD

**Discussion by choppsv1** [CESNET/libyang#1521] [Q&A]
With the following commit 1ccbf543f4d5d46ed6dd6690e98737928a47ee71

my FRR unit tests are failing in the following way:
```
2021/04/22 10:07:35 libyang: Invalid argument parent || module || ext (lys_getnext_()).
2021/04/22 10:07:35 libyang: Not found node "lib" in path. (Schema location /frr-routing:routing/control-plane-protocols/control-plane-protocol/frr-bgp:bgp/global/afi-safis/afi-safi/ipv4-flowspec/flow-spec-config/interface.)
2021/04/22 10:07:35 [Y3P23-A91R3][EC 100663318] yang_module_load: failed to load data model: frr-bgp
``

I realize this isn't a lot of info, but I was wondering if there is something I should look for based on that commit?

**Comment by michalvasko** [comment]
I do not think looking at the commit will help, it was one of those that seem fairly small but it significantly changed the way modules are compiled. Best would be if you could provide a simple example to reproduce it and it can be fixed and added to the tests.

**Reply by choppsv1** [reply]
Unfortunately BGP is the most complex of all the FRR modules (lots of submodules etc), it's going to take some time to get to the bottom of what's going on. For now I guess we'll have to freeze on the SHA before that commit.

**Reply by michalvasko** [reply]
Well, if I can help speed up the libyang release I think it is worth it even if you left it all to me and simply provided the full set of modules that cause the problem.

**Reply by michalvasko** [reply]
Please try the current version, I think this may have been fixed because the underlying bug was likely the same as in #1516.

**Reply by choppsv1** [reply]
Current version is still broken. If you'd really like to check it out here's how:
```
git clone https://github.com/LabNConsulting/frr.git
cd frr
git checkout chopps/ly2-triage
autoreconf -i
mkdir build
cd build
CPPFLAGS="$(pkg-config --cflags-only-I libyang)" ../configure --prefix=/usr --enable-dev-build --localstatedir=/var/run/frr --sbindir=/usr/lib/frr --sysconfdir=/etc/frr --enable-vtysh --enable-pimd --enable-sharpd --enable-snmp=agentx --enable-multipath=64 --enable-user=frr --enable-group=frr --enable-vty-group=frrvty --with-pkg-extra-version=-my-manual-build
make -j16
make check
```

The actual test that is failing can then be run:

```
$ tests/bgpd/.libs/test_peer_attr
2021/04/27 10:19:41 libyang: Invalid argument parent || module || ext (lys_getnext_()).
2021/04/27 10:19:41 libyang: Not found node "lib" in path. (Schema location /frr-routing:routing/control-plane-protocols/control-plane-protocol/frr-bgp:bgp/global/afi-safis/afi-safi/ipv4-flowspec/flow-spec-config/interface.)
2021/04/27 10:19:41 [Y3P23-A91R3][EC 100663318] yang_module_load: failed to load data model: frr-bgp
```

**Comment by choppsv1** [comment]
Is this change critical to the 2.0 release? Might I suggest if it is not then it should be backed out and attempted again after the release is done.

**Comment by choppsv1** [comment]
I've opened https://github.com/CESNET/libyang/issues/1528 to track this. Also included large debug output.

**Comment by choppsv1** [comment]
Now that this is working, hitting new bug with xpath not finding nodes :(

**Comment by choppsv1** [comment]
https://github.com/CESNET/libyang/issues/1532

**Comment by choppsv1** [comment]
https://github.com/CESNET/libyang/issues/1532
